src/torchmetrics/functional/regression/pearson.py

```diff
@@ -47,34 +48,56 @@ def _pearson_corrcoef_update(
         corr_xy: current covariance estimate between x and y tensor
         n_prior: current number of observed observations
         num_outputs: Number of outputs in multioutput setting
-
+        weights: weights associated with scores
     """
     # Data checking
     _check_same_shape(preds, target)
     _check_data_shape_to_num_outputs(preds, target, num_outputs)
-    n_obs = preds.shape[0]
+    if weights is not None:
+        _check_data_shape_to_weights(preds, weights)
+
+    n_obs = preds.shape[0] if weights is None else weights.sum()
     cond = n_prior.mean() > 0 or n_obs == 1
 
     if cond:
-        mx_new = (n_prior * mean_x + preds.sum(0)) / (n_prior + n_obs)
-        my_new = (n_prior * mean_y + target.sum(0)) / (n_prior + n_obs)
+        if weights is None:
+            mx_new = (n_prior * mean_x + preds.sum(0)) / (n_prior + n_obs)
+            my_new = (n_prior * mean_y + target.sum(0)) / (n_prior + n_obs)
+        else:
+            mx_new = (n_prior * mean_x + torch.matmul(weights, preds)) / (n_prior + n_obs)
+            my_new = (n_prior * mean_y + torch.matmul(weights, target)) / (n_prior + n_obs)
     else:
-        mx_new = preds.mean(0)
-        my_new = target.mean(0)
+        if weights is None:
+            mx_new = preds.mean(0)
+            my_new = target.mean(0)
+        else:
+            mx_new = torch.matmul(weights, preds) / weights.sum()
+            my_new = torch.matmul(weights, target) / weights.sum()
```

Comment:
Can we rather have have if `weights` is NOne to make it full of ones to have no effect but the code simpler